src/os_posix/os_dll.c

```diff
@@ -41,3 +59,62 @@ mcr_dll_sym (mcr_dll dll, const char *sym)
 }
 
 #endif
+
+#ifdef __APPLE__
+
+#include <mach-o/dyld.h>
+#include <mach-o/nlist.h>
+
+mcr_dll_path_result
+mcr_dll_path (mcr_dll dll)
+{
+   // Clear the three low bits of the module handle:
+   uintptr_t needle = ((uintptr_t) dll._native_handle & ~UINT64_C (0x3));
+   /// NOTE: Not thread safe. Is there a thread-safe way to do this?
```

Comment:
I think at least one reasonable protection here would be to handle a `dyld_name == NULL` result below.

Reply:
Maybe, but if `idx` can be invalidated between the loop beginning and `_dyld_get_image_name()`, then the returned pointer can also be invalidated between the call of `_dyld_get_image_name()` and `dlopen()` (Since `dyld` owns that pointer).

Reply:
Yeah, good point. I think what you have come up with in [1b84fed](https://github.com/mongodb/libmongocrypt/pull/263/commits/1b84fed369ac63373cb5d286e042c478e8bfd0db) seems like a pretty good solution.